packages/contracts-bedrock/src/safe/LivenessModule2.sol

```diff
@@ -0,0 +1,323 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Safe
+import { GnosisSafe as Safe } from "safe-contracts/GnosisSafe.sol";
+import { Enum } from "safe-contracts/common/Enum.sol";
+import { OwnerManager } from "safe-contracts/base/OwnerManager.sol";
+
+// Interfaces
+import { ISemver } from "interfaces/universal/ISemver.sol";
+
+/// @title LivenessModule2
+/// @notice This module allows challenge-based ownership transfer to a fallback owner
+///         when the Safe becomes unresponsive. The fallback owner can initiate a challenge,
+///         and if the Safe doesn't respond within the challenge period, ownership transfers
+///         to the fallback owner.
+/// @dev This is a singleton contract. To use it:
+///      1. The Safe must first enable this module using ModuleManager.enableModule()
+///      2. The Safe must then configure the module by calling configure() with params
+contract LivenessModule2 is ISemver {
+    /// @notice Configuration for a Safe's liveness module.
+    /// @custom:field livenessResponsePeriod The duration in seconds that Safe owners have to
+    ///                                      respond to a challenge.
+    /// @custom:field fallbackOwner The address that can initiate challenges and claim
+    ///                             ownership if the Safe is unresponsive.
+    struct ModuleConfig {
+        uint256 livenessResponsePeriod;
+        address fallbackOwner;
+    }
+
+    /// @notice Mapping from Safe address to its configuration.
+    mapping(address => ModuleConfig) public safeConfigs;
```

Comment:
To facilitate merging with the TimelockGuard, this should be `livenessSafeConfiguration`